nixos/tests/make-test-hardware.nix

```diff
@@ -0,0 +1,32 @@
+f: {
+  system ? builtins.currentSystem,
+  pkgs ? import ../.. { inherit system; },
+  ...
+} @ args:
+
+let
+  config = (if pkgs.lib.isFunction f then f (args // { inherit pkgs; inherit (pkgs) lib; }) else f);
+
+  sandboxPaths = config.sandboxPaths or [ "/sys" ];
+
+  sandboxPathsTests = builtins.map (path: "[[ ! -e '${path}' ]]") sandboxPaths;
+  sandboxPathsTest = pkgs.lib.concatStringsSep " || " sandboxPathsTests;
+  sandboxPathsList = pkgs.lib.concatStringsSep " " sandboxPaths;
+in
+pkgs.stdenv.mkDerivation (config // {
+  name = "test-run-${config.name}";
+
+  requiredSystemFeatures = [ "nixos-test" ];
+
+  buildCommand = ''
+    mkdir -p $out
+
+    if ${sandboxPathsTest}; then
+      echo 'Run this test as *root* with `--option extra-sandbox-paths '"'${sandboxPathsList}'"'` and a GPU'
+      exit 1
```

Comment:
I think one issue here is that this test will always show up as "failed" in hydra and ofborg, which is not so nice.

Reply:
If we are already give up on impurity could we not just have scripts that run outside the nix sandbox and people with the appropriate hardware would run those? Than we also don't need to remember all those sandbox flags.